tools/mdbook_preprocessor/mdbook_preprocessor.js

```diff
@@ -0,0 +1,151 @@
+// @ts-check
+/// <reference types="node" />
+
+/**
+ * @typedef {import("./types").Book} Book
+ * @typedef {import("./types").BookItem} BookItem
+ * @typedef {import("./types").Chapter} Chapter
+ *
+ * This file is an mdBook preprocessor that modifies the book JSON data.
+ * We use it to process code blocks with language "command-summary".
+ *
+ * @see processCommandSummary
+ *
+ * mdBook preprocessor documentation:
+ * https://rust-lang.github.io/mdBook/for_developers/preprocessors.html
+ */
+
+if (process.argv.length > 2) {
+  handleSupports();
+} else {
+  await handlePreprocess();
+}
+
+function handleSupports() {
+  if (process.argv[2] === "supports" && process.argv[3] === "html") {
+    process.exit(0);
+  }
+  process.exit(1);
+}
+
+/**
+ * An mdBook preprocessor receives JSON data `[context, book]` from stdin,
+ * modifies `book`, and writes `book` back to stdout.
+ *
+ * Example input:
+ * https://rust-lang.github.io/mdBook/for_developers/preprocessors.html#:~:text=mod%20test
+ */
+async function handlePreprocess() {
+  // Read from stdin
+  let stdin = "";
+  for await (const chunk of process.stdin) {
+    stdin += chunk;
+  }
+
+  // We don't care about the context. Only process the book.
+  const [, book] = JSON.parse(stdin);
+
+  processBook(book);
+
+  // Write to stdout
+  const output = JSON.stringify(book);
+  process.stdout.write(output + "\n");
+}
+
+/**
+ * @param {Book} book
+ */
+function processBook(book) {
+  for (const bookItem of book.sections) {
+    processBookItem(bookItem);
+  }
+}
+
+/**
+ * @param {BookItem} bookItem
+ */
+function processBookItem(bookItem) {
+  // bookItem is { "Chapter": Chapter }, "Separator", or { "PartTitle": string }
+  if (bookItem === "Separator") {
+    return;
+  }
+  // bookItem is { "Chapter": Chapter } or { "PartTitle": string }
+  if ("PartTitle" in bookItem) {
+    return;
+  }
+  // bookItem is { "Chapter": Chapter }
+
+  const chapter = bookItem.Chapter;
+  processChapter(chapter);
```

Comment:
Nitpick: you could inline the variable `chapter`.

```suggestion
  processChapter(bookItem.Chapter);
```